missing #include <unistd.h> for macOS in WordCompletion.cpp

when build in WordCompletion, got below error: 
fix it by add #include <unistd.h>

Undefined symbols for architecture x86_64:
  "isatty(int)", referenced from:
      word_init_buffer(yy_buffer_state*, __sFILE*, void*) in WordTokenizer.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/WordCompletion.dylib] Error 1
make[1]: *** [WordCompletion/CMakeFiles/WordCompletion.dir/all] Error 2
make: *** [all] Error 2

Readding my comments here as #1366 was closed.

It also fails linking on Mageia 6 x86_64 (a GNU/Linux distro, i.e. the issue is not macOS-specific):
```
[ 68%] Building CXX object codelite_make/CMakeFiles/codelite-make.dir/cl_make_generator_app.cpp.o
cd /home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/build/codelite_make && /bin/c++   -DGTK_USE_NATIVEBOOK=0 -DHAS_LIBCLANG_BRIEFCOMMENTS=1 -DINSTALL_DIR=\"/usr/share/codelite\" -DN$
CMakeFiles/WordCompletion.dir/WordTokenizer.cpp.o: In function `word_init_buffer(yy_buffer_state*, _IO_FILE*, void*)':
/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/build/WordCompletion/WordTokenizer.cpp:1599: undefined reference to `isatty(int)'
collect2: error: ld returned 1 exit status
WordCompletion/CMakeFiles/WordCompletion.dir/build.make:297: recipe for target 'lib/WordCompletion.so' failed
```

Note that 9.1 used to build fine.
Here is my RPM spec file: http://svnweb.mageia.org/packages/cauldron/codelite/current/SPECS/codelite.spec?view=markup
And the patches we're using (especially to unbundle external libraries): http://svnweb.mageia.org/packages/cauldron/codelite/current/SOURCES/

I tried this patch, but it introduces new issues:
```
diff --git a/WordCompletion/WordTokenizer.cpp b/WordCompletion/WordTokenizer.cpp
index 566cfa1..e5c795b 100644
--- a/WordCompletion/WordTokenizer.cpp
+++ b/WordCompletion/WordTokenizer.cpp
@@ -523,7 +523,10 @@ extern "C" int wordwrap(void*) { return 1; }
 #define ECHO
 #define P(s) fprintf(stderr, "%s\n", s)
 
+// Needed as per GH-1364
+#ifdef __WIN32
 #define YY_NO_UNISTD_H
+#endif
 #define YY_USER_ACTION  yycolumn += yyleng;
 
 /* options */
diff --git a/WordCompletion/WordTokenizer.l b/WordCompletion/WordTokenizer.l
index 89fef44..1f13cc6 100644
--- a/WordCompletion/WordTokenizer.l
+++ b/WordCompletion/WordTokenizer.l
@@ -13,7 +13,10 @@ extern "C" int yywrap(void*) { return 1; }
 #define ECHO
 #define P(s) fprintf(stderr, "%s\n", s)
 
+// Needed as per GH-1364
+#ifdef __WIN32
 #define YY_NO_UNISTD_H
+#endif
 #define YY_USER_ACTION  yycolumn += yyleng;
 
 %}
```

```
[ 68%] Building CXX object WordCompletion/CMakeFiles/WordCompletion.dir/WordTokenizer.cpp.o
cd /home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/build/WordCompletion && /bin/c++   -DGTK_USE_NATIVEBOOK=0 -DHAS_LIBCLANG_BRIEFCOMMENTS=1 -DINSTALL_DIR=\"/usr/share/codelite\" -DNDEBUG -DPLUGINS_DIR=\"/usr/lib64/codelite\" -DUSE_SFTP=1 -DWXUSINGDLL -DWXUSINGDLL_CL -DWXUSINGDLL_SDK -DWXUSINGDLL_WXSQLITE3 -DWordCompletion_EXPORTS -DYY_NEVER_INTERACTIVE=1 -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -isystem /usr/lib64/wx/include -isystem /usr/lib64/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -I/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/Plugin -I/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/sdk/wxsqlite3/include -I/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/CodeLite -I/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/PCH -I/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/Interfaces  -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4  -Wno-deprecated-declarations -std=c++11 -g -O2 -O2 -I/usr/include/wx-3.0 -fPIC -fPIC   -o CMakeFiles/WordCompletion.dir/WordTokenizer.cpp.o -c /home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/WordCompletion/WordTokenizer.cpp
In file included from /usr/include/features.h:365:0,
                 from /usr/include/stdio.h:27,
                 from WordTokenizer.cpp:19:
/usr/include/unistd.h:782:30: error: conflicting declaration of 'int isatty(int)' with 'C' linkage
 extern int isatty (int __fd) __THROW;
                              ^
In file included from WordTokenizer.cpp:519:0:
/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/WordCompletion/WordTokenizerAPI.h:6:12: note: previous declaration with 'C++' linkage
 extern int isatty(int);
            ^
In file included from /usr/include/features.h:365:0,
                 from /usr/include/stdio.h:27,
                 from WordTokenizer.cpp:19:
/usr/include/unistd.h:782:30: error: declaration of 'int isatty(int) throw ()' has a different exception specifier
 extern int isatty (int __fd) __THROW;
                              ^
In file included from WordTokenizer.cpp:519:0:
/home/akien/Mageia/Checkout/codelite/BUILD/codelite-9.2/WordCompletion/WordTokenizerAPI.h:6:12: error: from previous declaration 'int isatty(int)'
 extern int isatty(int);
            ^
```

--- a/WordCompletion/WordTokenizerAPI.h
+++ b/WordCompletion/WordTokenizerAPI.h
@@ -2,6 +2,10 @@
 #define WORD_TOKENIZER_API_H
 
 #include <wx/string.h>
+// Needed as per GH-1364
+#ifndef __WIN32
+#include <unistd.h>
+#endif
 
 extern int isatty(int);
 

